ood-portal-generator/sbin/update_ood_portal

```diff
@@ -27,45 +31,76 @@ while true; do
     case "$1" in
         -h|--help) usage ; exit 0 ; shift ;;
         -r|--rpm) RPM=1 ; shift ;;
+        -f|--force) FORCE=true ; shift ;;
         *) break ;;
     esac
 done
 
+
+if ! grep -q "$APACHE" $SUM 1>/dev/null 2>&1; then
+    echo "Generating Apache config checksum file: '${SUM}'"
+    sha256sum $APACHE > $SUM
+fi
+
 echo "Generating Apache config using YAML config: '${CONFIG}'"
 
-NEW_APACHE="$("${BIN}" -c "${CONFIG}")"
-if ! cmp -s <(echo "${NEW_APACHE}") "${APACHE}"; then
-  if [[ -f "${APACHE}" ]]; then
-    BAK_APACHE="${APACHE}.$(date '+%Y%m%dT%H%M%S')"
-    echo "Backing up previous Apache config to: '${BAK_APACHE}'"
-    mv "${APACHE}" "${BAK_APACHE}"
+NEW_APACHE=$(mktemp)
+"${BIN}" -c "${CONFIG}" > $NEW_APACHE
+
+if sha256sum --quiet --status --check $SUM 1>/dev/null 2>&1; then
+    REPLACE=true
+else
+    REPLACE=false
+fi
+if $FORCE; then
+    REPLACE=true
+fi
+if ! cmp -s "${NEW_APACHE}" "${APACHE}"; then
+  if $REPLACE; then
+    if [[ -f "${APACHE}" ]]; then
+      BAK_APACHE="${APACHE}.$(date '+%Y%m%dT%H%M%S')"
+      echo "Backing up previous Apache config to: '${BAK_APACHE}'"
+      mv "${APACHE}" "${BAK_APACHE}"
+    fi
+    echo "Generating new Apache config at: '${APACHE}'"
+    cat "${NEW_APACHE}" > "${APACHE}"
+    echo "Generating Apache config checksum file: '${SUM}'"
+    sha256sum $APACHE > $SUM
+    ret=0
+    CHANGED=true
+  else
+    echo "WARNING: Checksum of ${APACHE} does not match previous value, not replacing."
+    echo "Generating new Apache config at: '${APACHE}.new'"
+    cat "${NEW_APACHE}" > "${APACHE}.new"
```

Comment:
We're telling folks this .new file exists, but we delete it below.

Reply:
Where do you see the delete? The only file we remove is the temporary file that is used to write the .new file.